### DISCUSSION TITLE
Steady State for rotated TFI

### DISCUSSION THREAD

**Discussion by skothegh** [netket/netket#791] [Q&A] [answered]
Hi there,

Thanks for all the help so far with undererstanding NetKet and NNQS in general!

I am toying around with the 1D Ising Steady State example (https://github.com/netket/netket/blob/master/Examples/DissipativeIsing1d/ising1d.py), except that the interaction is Sx-Sx and the field is given by Sz. The jump operators are still Sm. The system parameters are V = 4, g = 0. Both in units of the dissipation strength. Apart from that I increased the system size to L = 16 and used Adam instead of Sgd.

It seems the network has some trouble converging towards the steady state. I have tried several combinations of meta parameters but nothing seems to improve the convergence by much.

Is the SxSx interaction with Sm-dissipation generally difficult for RBMs, or is is there a way to improve convergence in this case that I missed for lack of experience and intuition?

All the best and thanks for the awesome library! It makes getting into this topic a good deal more accessible!

**Comment by PhilipVinc** [comment, marked as answer]
This model is tough for the NDM for a series of reasons in the crossover regime from pure to mixed state (g \approx 1-2).

For `g=0` the steady state is very simple to determine. I never had problems doing it so it surprises me a bit?

You need to push a bit the mixed part of the NDM (beta) and the pure (alfa). Also might need to increase samples.
Can you share a plot?

**Reply by skothegh** [reply]
Hi thanks for the reply! I don't have a plot at hand but I can produce one now. I will post here once its done. 

I thought so as well. At least with my MPS code I can get it done with no problem at all. But RBMs are different beasts and I barely understand them half as well as I would like to, to be honest.

So far I have tried anything between 3000 and 8000 samples at a learning rate of 0.01, although always for beta=alpha=1. If I increase alpha and beta, should I also change any other parameter such as the learning rate/ number of iterations/ number of samples?

**Reply by skothegh** [reply]
Here is a plot. This is a wee bit lower than usually but it still goes horizontal pretty early. And from there it barely gets any better. 
L = 16
J = k = 1
h = 0

Nsample = 5000
eta = 0.01
alpha = beta = 1
Niter = 1000
Diag_shift = 0.01

I used 
![plot](https://user-images.githubusercontent.com/73349739/123838005-62c2e080-d903-11eb-9d21-ea16bd701d33.png)

**Reply by PhilipVinc** [reply]
Did you try with sr?

**Reply by skothegh** [reply]
Actually yes, that was a leftover from when I used SGD an played with different optimizers. I took it out now and it produces a similar result.

**Reply by PhilipVinc** [reply]
I tried quickly just now (a 20 seconds simulation) with 16 sites, g=0, and get what I'd expect.

(I'm plotting LdagL on the y axis, iteration # on the x axis)
![Figure_1](https://user-images.githubusercontent.com/2407108/123876730-1ea00200-d93c-11eb-826d-363e903eeb91.png)

You can find the code [here](https://gist.github.com/PhilipVinc/20fd6b09012d6a9d42151708fbd380f1)

**Reply by PhilipVinc** [reply]
For `V=4` just set 
```
vs.init_parameters(nk.nn.initializers.normal(stddev=0.001))
```
and you'll find roughly the same curve.

**Reply by skothegh** [reply]
Thats interesting. When I copy-paste that code and simply change 
L = 16, V = 4, g = 0, stddev=0.001, as well as Sx*Sx-> Sz*Sz, I get this: 

Edit: When I run your code directly I get an unexpected keyword error: "SR got an unexpected keyword argument 'qgt'"

![plot3](https://user-images.githubusercontent.com/73349739/123927294-9bef6500-d984-11eb-91e3-257edfa52af5.png)

**Reply by PhilipVinc** [reply]
you are on an older beta version. The QGT stuff got added more recently.

By the way, I noticed only later the interaction along the X direction. 
That is sure to wreak havoc. I know that particular model/ansatz to have issues encoding the correlations along the X directions. Putting the interactions along the same directions will surely degrade the encoding performance.

**Reply by skothegh** [reply]
Yea, thats where the problem lies. For work I need the SxSx interaction, at least short term. I think I will just have to crank up alpha and beta and hope for the best, at this point. Thanks for all the help, though!

Edit. When will the beta version become available?

**Reply by PhilipVinc** [reply]
The latest beta version (with the qgt argument and a reworked SR interface to be more general) is available since [netket v3.0b2](https://github.com/netket/netket/releases) . 
You are probably on a beta1 version if you don't have those kwargs.

To update just do `pip install -U --pre netket` and, since sometimes he does not update dependencies that way, also update
`pip install -U jax jaxlib plum-dispatch flax optax mpi4jax`.

We will release a beta 3 in ~ a week with a few performance improvements. The final 3.0 release is slated for late august.

**Reply by skothegh** [reply]
Sweet, thanks a lot!
